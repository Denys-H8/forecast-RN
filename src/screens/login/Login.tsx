import React, { useEffect, useState } from 'react';
import { View } from 'react-native';
import * as Keychain from 'react-native-keychain';

import { styles } from './login.styles';
import Input from './components/input';
import Button from './components/button';
import constants from 'src/constants';
import { useDispatch, useSelector } from 'react-redux';
import { loginActions } from 'src/store/rootActions';
import * as LOGIN_SELECTORS from '../../store/login/selectors';

// Email: root@gmail.com
// Password: passroot

const Login: React.FC = () => {
  const [btnDisabled, setBtnDisabled] = useState(true);

  const dispatch = useDispatch();

  useEffect(() => {
    const getEmail = async () => {
      try {
        const credentials: false | Keychain.UserCredentials = await Keychain.getGenericPassword();
        if (credentials) dispatch(loginActions.changedEmail(credentials?.password));
      } catch (error) {
        console.log("Keychain couldn't be accessed!", error);
      }
    };
    getEmail();
  }, []);

  const email = useSelector(LOGIN_SELECTORS.email);
  const password = useSelector(LOGIN_SELECTORS.password);
  const emailError = useSelector(LOGIN_SELECTORS.emailError);
  const passError = useSelector(LOGIN_SELECTORS.passError);

  const emailValidation = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;

  const onChangeEmail = (val: string) => {
    dispatch(loginActions.changedEmail(val));

    emailValidation.test(val) && password.length > 7 ? setBtnDisabled(false) : setBtnDisabled(true);
  };

  const onChangePassword = (val: string) => {
    dispatch(loginActions.changedPassword(val));

    emailValidation.test(email) && val.length > 7 ? setBtnDisabled(false) : setBtnDisabled(true);
  };

  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <View style={styles.topContainer}>
          <Input
            value={email}
            error={emailError}
            placeholder="Email"
            onChangeText={(val) => onChangeEmail(val)}
            placeholderTextColor={constants.MAIN_COLOR}
            iconName="account-circle"
          />
        </View>
        <View style={styles.bottomContainer}>
          <Input
            value={password}
            error={passError}
            placeholder="Password"
            onChangeText={(val) => onChangePassword(val)}
            placeholderTextColor={constants.MAIN_COLOR}
            iconName="lock"
            secureTextEntry
          />
        </View>
        <Button onPress={() => dispatch(loginActions.login())} isDisabled={btnDisabled} />
      </View>
    </View>
  );
};

export default Login;
