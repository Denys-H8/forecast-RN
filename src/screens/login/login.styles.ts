import { StyleSheet } from 'react-native';

import constants from 'src/constants';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: constants.BG_COLOR,
  },
  content: {
    flex: 1,
    paddingVertical: 50,
    paddingHorizontal: 32,
  },
  topContainer: {
    flex: 1,
    justifyContent: 'flex-end',
    paddingBottom: 12,
  },
  bottomContainer: {
    flex: 1,
    paddingTop: 12,
  },
  text: {
    color: constants.ACTIVE_TEXT_COLOR,
    fontSize: 22,
  }
});
