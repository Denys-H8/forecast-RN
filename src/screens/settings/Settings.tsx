import React from 'react';
import { View, TouchableOpacity } from 'react-native';

import { CustomText } from 'src/components';
import { styles } from './settings.styles';
import InputField from './components/inputField';
import TemperatureType from './components/temperatureType';
import Slider from './components/slider';
import { useDispatch, useSelector } from 'react-redux';
import { loginActions, settingsActions } from 'src/store/rootActions';
import * as SETTINGS_SELECTORS from '../../store/settings/selectors';

const Settings: React.FC = () => {
  const dispatch = useDispatch();

  const dayPeriod = useSelector(SETTINGS_SELECTORS.dayPeriod);
  const name = useSelector(SETTINGS_SELECTORS.name);
  const age = useSelector(SETTINGS_SELECTORS.age);
  const updateTime = useSelector(SETTINGS_SELECTORS.updateTime);
  const temperatureScale = useSelector(SETTINGS_SELECTORS.temperatureScale);

  const handleChangeTempScale = (temperatureScale: string) => {
    dispatch(settingsActions.changedTemperatureScale(temperatureScale));
  };

  return (
    <View style={styles.container}>
      <View>
        <View style={styles.section}>
          <InputField
            placeholder="Name"
            value={name}
            onChangeText={(value) => dispatch(settingsActions.changedName(value))}
          />
          <InputField
            placeholder="Age"
            value={age.toString()}
            onChangeText={(value) => dispatch(settingsActions.changedAge(value))}
          />
        </View>
        <View style={styles.section}>
          <Slider
            title="Show weather for"
            text={`${dayPeriod} days`}
            minimumValue={1}
            maximumValue={5}
            value={dayPeriod}
            onChange={(value) => {
              dispatch(settingsActions.changedDayPeriod(value));
            }}
          />
        </View>
        <View style={styles.section}>
          <Slider
            title="Update weather every"
            text={`${updateTime} min`}
            minimumValue={5}
            maximumValue={60}
            value={updateTime}
            onChange={(value) => {
              dispatch(settingsActions.changedUpdateTime(value));
            }}
          />
        </View>
        <View style={[styles.section, styles.tempSection]}>
          <TouchableOpacity onPress={() => handleChangeTempScale('metric')} testID="Settings.Scale.Metric">
            <TemperatureType type="metric" isActive={temperatureScale} />
          </TouchableOpacity>
          <View style={styles.delimiter} />
          <TouchableOpacity onPress={() => handleChangeTempScale('imperial')} testID="Settings.Scale.Imperial">
            <TemperatureType type="imperial" isActive={temperatureScale} />
          </TouchableOpacity>
        </View>
        <View style={[styles.section, { borderBottomWidth: 0 }]}>
          <TouchableOpacity
            onPress={() => {
              dispatch(loginActions.logout());
            }}
            testID="Settings.Logout"
          >
            <CustomText isTitle style={styles.logoutBtn}>
              Logout
            </CustomText>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
};

export default Settings;
