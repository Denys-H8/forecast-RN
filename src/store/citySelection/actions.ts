import * as ACTION from './actionTypes';
import { TGroupWeather, TData, TCity } from './reducer';

export const queryChanged = (query: string, data: TData[]) => ({
  type: ACTION.QUERY_CHANGED,
  payload: { query, data },
});
export const getWeatherForCities = () => ({ type: ACTION.GET_WEATHER_FOR_CITIES });
export const getWeatherForCitiesSucess = (data: TGroupWeather) => ({
  type: ACTION.GET_WEATHER_FOR_CITIES_SUCESS,
  payload: data,
});
export const resetQuery = () => ({ type: ACTION.RESET_QUERY });
export const changeCity = (city: TCity) => ({
  type: ACTION.CHANGE_CITY,
  payload: city,
});
export const pressedCity = (city: TCity) => ({
  type: ACTION.PRESSED_CITY,
  payload: city,
});
export const setLoading = (value: boolean) => ({ type: ACTION.SET_LOADING, payload: value });
