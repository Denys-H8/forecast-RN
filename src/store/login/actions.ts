import * as ACTION from './actionTypes';
import { TUser } from './reducer';

export const login = () => ({ type: ACTION.LOGIN });
export const logout = () => ({ type: ACTION.LOGOUT });
export const logoutSuccess = () => ({ type: ACTION.LOGOUT_SUCCESS });
export const loginStarted = () => ({ type: ACTION.LOGIN_STARTED });
export const loginSuccess = (user: TUser) => ({ type: ACTION.LOGIN_SUCCESS, payload: user });
export const changedEmail = (email: string | false) => ({ type: ACTION.CHANGED_EMAIL, payload: email });
export const changedPassword = (password: string) => ({ type: ACTION.CHANGED_PASSWORD, payload: password });
export const failedEmail = () => ({ type: ACTION.FAILED_EMAIL });
export const failedPassword = () => ({ type: ACTION.FAILED_PASSWORD });
export const startLoading = () => ({ type: ACTION.START_LOADING });
export const loading = (value: boolean) => ({ type: ACTION.LOADING_LOGIN, payload: value });
