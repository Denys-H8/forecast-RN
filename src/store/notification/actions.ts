import { FirebaseMessagingTypes } from '@react-native-firebase/messaging';
import * as ACTION from './actionTypes';

export const notification = (routeName: string) => ({ type: ACTION.NOTIFICATION, payload: routeName });
export const addNotification = (notification: FirebaseMessagingTypes.RemoteMessage) => ({
  type: ACTION.ADD_NOTIFICATION,
  payload: notification,
});
export const removeNotification = (notification: FirebaseMessagingTypes.RemoteMessage) => ({
  type: ACTION.REMOVE_NOTIFICATION,
  payload: notification,
});
