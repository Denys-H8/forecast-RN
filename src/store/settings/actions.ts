import * as ACTION from './actionTypes';

export const changedName = (name: string) => ({ type: ACTION.CHANGED_NAME, payload: name });
export const changedAge = (age: string) => ({ type: ACTION.CHANGED_AGE, payload: age });
export const changedDayPeriod = (days: number) => ({ type: ACTION.CHANGED_DAY_PERIOD, payload: days });
export const changedUpdateTime = (time: number) => ({ type: ACTION.CHANGED_UPDATE_TIME, payload: time });
export const changedTemperatureScale = (scale: string) => ({
  type: ACTION.CHANGED_TEMPERATURE_SCALE,
  payload: scale,
});
export const setTemperatureScale = (scale: string) => ({
  type: ACTION.SET_TEMPERATURE_SCALE,
  payload: scale,
});
