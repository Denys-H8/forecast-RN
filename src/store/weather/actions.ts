import * as ACTION from './actionTypes';
import { TWeather, TWeatherStats } from './reducer';

export const getWeather = () => ({ type: ACTION.GET_WEATHER });
export const getWeatherSuccess = (weatherData: TWeatherData) => ({
  type: ACTION.GET_WEATHER_SUCCESS,
  payload: weatherData,
});
export const loading = (value: boolean) => ({ type: ACTION.LOADING, payload: value });
export const lateNavigation = (routeName: string | undefined) => ({ type: ACTION.LATE_NAVIGATION, payload: routeName });
export const resetLateNavigation = () => ({ type: ACTION.RESET_LATE_NAVIGATION });

export type TWeatherData = {
  weather: TWeather;
  weatherStats: TWeatherStats;
};
