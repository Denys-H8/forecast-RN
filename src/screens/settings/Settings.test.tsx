import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Settings from './Settings';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import {
  changedAge,
  changedDayPeriod,
  changedName,
  changedTemperatureScale,
  changedUpdateTime,
} from 'src/store/settings/actions';
import { logout } from 'src/store/login/actions';

const mockStore = configureStore([]);

const store = mockStore({
  settingsReducer: {
    dayPeriod: 4,
    name: 'Name',
    age: 19,
    updateTime: 15,
    temperatureScale: 'metric',
  },
});

describe('Forecast ', () => {
  test('should render correctly', () => {
    const { toJSON } = render(
      <Provider store={store}>
        <Settings />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test('should handle TempScale change metric', () => {
    store.dispatch = jest.fn();

    const { getByTestId } = render(
      <Provider store={store}>
        <Settings />
      </Provider>,
    );

    fireEvent.press(getByTestId('Settings.Scale.Metric'));
    expect(store.dispatch).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledWith(changedTemperatureScale('metric'));
  });

  test('should handle TempScale change imperial', () => {
    store.dispatch = jest.fn();

    const { getByTestId } = render(
      <Provider store={store}>
        <Settings />
      </Provider>,
    );

    fireEvent.press(getByTestId('Settings.Scale.Imperial'));
    expect(store.dispatch).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledWith(changedTemperatureScale('imperial'));
  });

  test('should handle logout', () => {
    store.dispatch = jest.fn();

    const { getByTestId } = render(
      <Provider store={store}>
        <Settings />
      </Provider>,
    );

    fireEvent.press(getByTestId('Settings.Logout'));
    expect(store.dispatch).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledWith(logout());
  });

  test('should handle change name', () => {
    store.dispatch = jest.fn();

    const newName = 'newName';
    const { getAllByTestId } = render(
      <Provider store={store}>
        <Settings />
      </Provider>,
    );
    fireEvent.changeText(getAllByTestId('InputField.TextInput')[0], newName);

    expect(store.dispatch).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledWith(changedName(newName));
  });

  test('should handle change age', () => {
    store.dispatch = jest.fn();

    const newAge = '20';
    const { getAllByTestId } = render(
      <Provider store={store}>
        <Settings />
      </Provider>,
    );

    fireEvent.changeText(getAllByTestId('InputField.TextInput')[1], newAge);

    expect(store.dispatch).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledWith(changedAge(newAge));
  });

  test('should handle change weather period', () => {
    store.dispatch = jest.fn();

    const newValue = 2;
    const { getAllByTestId } = render(
      <Provider store={store}>
        <Settings />
      </Provider>,
    );

    fireEvent(getAllByTestId('Slider.Slider')[0], 'onValueChange', newValue);

    expect(store.dispatch).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledWith(changedDayPeriod(newValue));
  });

  test('should handle change update period', () => {
    store.dispatch = jest.fn();

    const newValue = 20;
    const { getAllByTestId } = render(
      <Provider store={store}>
        <Settings />
      </Provider>,
    );

    fireEvent(getAllByTestId('Slider.Slider')[1], 'onValueChange', newValue);

    expect(store.dispatch).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledWith(changedUpdateTime(newValue));
  });
});
