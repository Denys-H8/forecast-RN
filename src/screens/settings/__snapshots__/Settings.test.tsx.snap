// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast  should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#191C39",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "borderBottomColor": "#8B8EA5",
          "borderBottomWidth": 2,
          "paddingBottom": 6,
          "paddingHorizontal": 32,
          "paddingTop": 12,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#191C39",
            "borderBottomColor": "#fff",
            "borderBottomWidth": 2,
            "marginBottom": 26,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Name"
          placeholderTextColor="#8B8EA5"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#fff",
              "fontFamily": "WorkSans-Regular.ttf",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
          testID="InputField.TextInput"
          underlineColorAndroid="transparent"
          value="Name"
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#191C39",
            "borderBottomColor": "#fff",
            "borderBottomWidth": 2,
            "marginBottom": 26,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Age"
          placeholderTextColor="#8B8EA5"
          rejectResponderTermination={true}
          style={
            Object {
              "color": "#fff",
              "fontFamily": "WorkSans-Regular.ttf",
              "fontSize": 16,
              "lineHeight": 18,
            }
          }
          testID="InputField.TextInput"
          underlineColorAndroid="transparent"
          value="19"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#8B8EA5",
          "borderBottomWidth": 2,
          "paddingBottom": 6,
          "paddingHorizontal": 32,
          "paddingTop": 12,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "WorkSans-Regular",
                },
                Object {
                  "color": "#fff",
                  "fontFamily": "WorkSans-SemiBold",
                  "fontSize": 22,
                  "lineHeight": 36,
                },
              ]
            }
          >
            Show weather for
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "WorkSans-Regular",
                },
                Object {
                  "color": "#fff",
                  "fontSize": 22,
                  "lineHeight": 36,
                },
              ]
            }
          >
            4 days
          </Text>
        </View>
        <RNCSlider
          disabled={false}
          enabled={true}
          inverted={false}
          maximumTrackTintColor="#8B8EA5"
          maximumValue={5}
          minimumTrackTintColor="#7B61FF"
          minimumValue={1}
          onChange={[Function]}
          onRNCSliderSlidingComplete={null}
          onRNCSliderSlidingStart={null}
          onRNCSliderValueChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          step={1}
          style={
            Array [
              Object {
                "height": 40,
              },
              Object {
                "height": 40,
              },
            ]
          }
          testID="Slider.Slider"
          thumbImage={null}
          value={4}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#8B8EA5",
          "borderBottomWidth": 2,
          "paddingBottom": 6,
          "paddingHorizontal": 32,
          "paddingTop": 12,
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "WorkSans-Regular",
                },
                Object {
                  "color": "#fff",
                  "fontFamily": "WorkSans-SemiBold",
                  "fontSize": 22,
                  "lineHeight": 36,
                },
              ]
            }
          >
            Update weather every
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "WorkSans-Regular",
                },
                Object {
                  "color": "#fff",
                  "fontSize": 22,
                  "lineHeight": 36,
                },
              ]
            }
          >
            15 min
          </Text>
        </View>
        <RNCSlider
          disabled={false}
          enabled={true}
          inverted={false}
          maximumTrackTintColor="#8B8EA5"
          maximumValue={60}
          minimumTrackTintColor="#7B61FF"
          minimumValue={5}
          onChange={[Function]}
          onRNCSliderSlidingComplete={null}
          onRNCSliderSlidingStart={null}
          onRNCSliderValueChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          step={1}
          style={
            Array [
              Object {
                "height": 40,
              },
              Object {
                "height": 40,
              },
            ]
          }
          testID="Slider.Slider"
          thumbImage={null}
          value={15}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#8B8EA5",
            "borderBottomWidth": 2,
            "paddingBottom": 6,
            "paddingHorizontal": 32,
            "paddingTop": 12,
          },
          Object {
            "flexDirection": "row",
            "paddingBottom": 18,
            "paddingTop": 18,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="Settings.Scale.Metric"
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "WorkSans-Regular",
                  },
                  Array [
                    Object {
                      "color": "#8B8EA5",
                      "fontFamily": "WorkSans-SemiBold",
                      "fontSize": 16,
                      "lineHeight": 18,
                    },
                    Object {
                      "color": "#fff",
                    },
                  ],
                ]
              }
            >
              0
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "WorkSans-Regular",
                  },
                  Array [
                    Object {
                      "color": "#8B8EA5",
                      "fontFamily": "WorkSans-SemiBold",
                      "fontSize": 28,
                      "lineHeight": 30,
                    },
                    Object {
                      "color": "#fff",
                    },
                  ],
                ]
              }
            >
              C
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#8B8EA5",
            "marginHorizontal": 10,
            "width": 2,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="Settings.Scale.Imperial"
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "WorkSans-Regular",
                  },
                  Object {
                    "color": "#8B8EA5",
                    "fontFamily": "WorkSans-SemiBold",
                    "fontSize": 16,
                    "lineHeight": 18,
                  },
                ]
              }
            >
              0
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "WorkSans-Regular",
                  },
                  Object {
                    "color": "#8B8EA5",
                    "fontFamily": "WorkSans-SemiBold",
                    "fontSize": 28,
                    "lineHeight": 30,
                  },
                ]
              }
            >
              F
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#8B8EA5",
            "borderBottomWidth": 2,
            "paddingBottom": 6,
            "paddingHorizontal": 32,
            "paddingTop": 12,
          },
          Object {
            "borderBottomWidth": 0,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="Settings.Logout"
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "WorkSans-ExtraBold",
                "fontSize": 52,
                "lineHeight": 61,
              },
              Object {
                "color": "#fff",
                "fontSize": 28,
                "lineHeight": 30,
                "marginTop": 18,
              },
            ]
          }
        >
          Logout
        </Text>
      </View>
    </View>
  </View>
</View>
`;
